src/pages/index.js

```diff
@@ -1,130 +1,223 @@
+    Constants.configuration,
+    Constants.addForm
+  );
+  const avatarValidator = new FormValidator(
+    Constants.configuration,
+    Constants.avatarForm
+  );
+  const user = new UserInfo(
+    Constants.profileName,
+    Constants.profileDescription,
+    Constants.avatarPhoto
+  );
 
   editValidator.enableValidation();
   addValidator.enableValidation();
-
-  const profile = document.querySelector("#profile");
-  const profileEditButton = profile.querySelector("#profile__edit_button");
-  const profileAddButton = profile.querySelector("#profile__add_button");
-  const profileName = profile.querySelector("#profile__name");
-  const profileDescription = profile.querySelector("#profile__description");
+  avatarValidator.enableValidation();
 
   const imagePopup = new PopupWithImage("#picture-modal");
 
   const editPopup = new PopupWithForm(
-    `#${editForm.parentElement.id}`,
+    `#${Constants.editForm.parentElement.id}`,
     handleProfileEditSubmit
   );
   const addPopup = new PopupWithForm(
-    `#${addForm.parentElement.id}`,
+    `#${Constants.addForm.parentElement.id}`,
     handleProfileAddSubmit
   );
-  const userInfo = new UserInfo(profileName, profileDescription);
 
   imagePopup.setEventListeners();
 
+  api
+    .getProfileInfo()
+    .then((results) => {
+      user.setUserInfo({ name: results.name, about: results.about });
+      user.setAvatar(results.avatar);
+    })
+    .catch((error) => {
+      console.error("Caught an error: ", error.message);
+    });
+
   //functions
+  const deletePopup = new ConfirmationPopup("#confirmation-modal");
+  deletePopup.setEventListeners();
+
+  function addCard(name, description, template, id) {
+    const items = [{ name: name, description: description }];
+    const modelCard = new Card(
+      name,
+      description,
+      template,
+      imageClickHandler,
+      (card) => {
+        deletePopup.open();
+        deletePopup.setDefaultSaveButton();
+        deletePopup.setSubmitAction(() => {
+          api
+            .deleteCard(card.id)
+            .then(() => {
+              card.deleteCard();
+            })
+            .then(() => {
+              deletePopup.updateSaveButton();
+            })
+            .then(() => {
+              deletePopup.close();
+            })
+            .catch((error) => {
+              console.error("Caught an error: ", error.message);
+            });
+        });
+      },
+      toggleLikeButton,
+      initialCardsInfo,
+      id
+    );
+    cardsSection.items = items;
+    cardsSection.addItem(modelCard.generateCard());
+  }
 
   function createCard(name, description, template) {
-    const modelCard = new Card(name, description, template, imageClickHandler);
-    cardsSection.addItem(modelCard.addCard());
+    return api
+      .addCardInfo(name, description, (name, link, id) => {
+        addCard(name, link, template, id);
+      })
+      .then(
+        addPopup.close()
+      )
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


```js
      .then(() => addPopup.close())


```